apps/balancer-tools/src/app/apr/(utils)/fetchPoolSnapshots.ts

```diff
@@ -0,0 +1,43 @@
+import { PoolSnapshotInRangeQuery } from "@bleu-balancer-tools/gql/src/balancer/__generated__/Ethereum.server";
+
+import { pools, poolsWithoutCache } from "#/lib/gql/server";
+
+import { dateToEpoch, generateDateRange } from "../api/(utils)/date";
+
+export async function fetchPoolSnapshots(
+  to: number,
+  extendedFrom: number,
+  network: string,
+  poolId: string,
+): Promise<PoolSnapshotInRangeQuery> {
+  let res: PoolSnapshotInRangeQuery;
+
+  const now = new Date();
+  const currentDateUTC = dateToEpoch(
+    new Date(
+      Date.UTC(
+        now.getUTCFullYear(),
+        now.getUTCMonth(),
+        now.getUTCDate(),
+        0,
+        0,
+        0,
+        0,
+      ),
+    ),
+  );
+
+  if (to === currentDateUTC) {
+    res = await poolsWithoutCache.gql(network).poolSnapshotInRange({
+      poolId,
+      timestamp: generateDateRange(extendedFrom, to),
+    });
+  } else {
+    res = await pools.gql(network).poolSnapshotInRange({
+      poolId,
+      timestamp: generateDateRange(extendedFrom, to),
+    });
+  }
```

Comment:
not the biggest fan of these... but it gets the job done. Happy to hear other solutions

Reply:
you could do something like 
```
strategy = to === currentDateUTC ? poolsWithoutCache : pools;
const res = await strategy.gql(network).poolSnapshotInRange({
  poolId,
  timestamp: generateDateRange(extendedFrom, to),
});
```